gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -1267,6 +1270,25 @@ else if (SQLState.INTERNAL_SKIP_ORLIST_OPTIMIZATION.equals(messgId)) {
 		return preparedStmt;
 	}
 
+	public boolean invalidQueryOnColumnTable(LanguageConnectionContext _lcc,
+			DMLQueryInfo qi) {
+		boolean isColumnTable = SnappyActivation.isColumnTable(qi);
+		// check subqueries
+		List<SubQueryInfo> subQinfoList = qi.getSubqueryInfoList();
+		if (!isColumnTable && subQinfoList.size() > 0) {
+			for (SubQueryInfo sq : subQinfoList) {
+				isColumnTable = SnappyActivation.isColumnTable(sq);
+				if(isColumnTable)
+					break;
+			}
+		}
+		// additional step for insertAsSubSelect
+		if (!isColumnTable && qi.isInsertAsSubSelect()) {
+			isColumnTable = SnappyActivation.isColumnTable(((InsertQueryInfo)qi).getSubSelectQueryInfo());
```

Comment:
Indentation may be corrected here by taking the line more towards right. Please check wherever applicable. Noticed above also.

Reply:
Will correct the indentation

Reply:
Not sure why you especially handled insertAsSubSelect. Shouldn't similar treatment be given for put-into-select?

Reply:
During testing found that insertAsSubSelect was initializing its fields via a call to handleInsertAndInsertSubSelect and was saving the subSelectInfo in its member param InsertQueryInfo#subSelectInfo so had to add this handling for "insert into <> select query". The InsertQueryInfo name appears to be a misnomer here as it appears its to be used only for "insert into <> select query" only and not for all inserts. put-into-select is also taken care by this change will add a test for it.